### PR TITLE
clang-formating and argument flags for main

### PR DESCRIPTION
**Pull request by nimertz**
Small PR for making benchmarking and collaborating easier.
* Formatted all source files to the clang-format we decided upon
    *  Required that some comments and TODO's where moved above the statement rather than to the right of it. Sorry.
* Basic documentation for ANN-Benchmark script added to readme
* Quickly added Argument flags to main - see below for usage in VTune
    * L = -l , metric = -m, hdf5 file = -f <Path> , NN to find = -k , b = -b 
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@ env
 # User-specific files
 *.out
 ann-benchmarks/*
+eCP/cmake-build-debug/
```
Comment (mortenskoett) [inline .gitignore:15]: Maybe ask your IDE to use eCP/build as build dir. Just a thought.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ build.sh
 This will build the project in Release mode. To change this see `configure.sh` script. Currently only GNU/GCC is known to be supported.
 
 ## How to run eCP in ANN-Benchmarks
+### Running it manually
 1. Go to `./ann_benchmarks/install/Dockerfile.ecp` and change the line describing what repository is cloned when the eCP docker container 
 is build by Ann-Benchmarks (AB).
 Make it point to the current working branch e.g. `git clone -b <branch_name> <url_to_repo> eCP`, where 'eCP will be local name of repo dir to help build scripts.
```
Comment (mortenskoett) [inline README.md:31]: Is this still possible?

File: README.md [hunk 3]
```diff
@@ -38,6 +39,16 @@ from the designated branch from earlier.
 
 4. Goto AB directory and follow the guide to run tests found at `https://github.com/erikbern/ann-benchmarks`.
 
+### Using the provided run script
```
Comment (mortenskoett) [inline README.md:42]: Nice!

File: README.md [hunk 4]
```diff
@@ -38,6 +39,16 @@ from the designated branch from earlier.
...
+From within the eCP/scripts folder run the ann-benchmark.sh script, it takes arbitrarily many arguments in a array-style format depending on the amount of experiments the user wants to run. An example configuration could be ```sudo ./ann-benchmark.sh [abc123,foo,random-xs-20-euclidean,false] [def456,bar,random-xs-20-euclidean,true]``` corresponding to running the code at commit id abc123 on the random-xs-20-euclidean dataset without early halting and naming it foo whereafter commit id def456 would be run on the same dataset with early halting enabled and under the name bar. Observe that there cannot be any spaces after each comma.
+
+It supports the following flags:
+* -d (--delete) delete the current local ann-benchmarks repository if it exists
+* -l (--local) run ann-benchmarks in local mode (`https://github.com/erikbern/ann-benchmarks`)
```
Comment (mortenskoett) [inline README.md:47]: Excellent

File: README.md [hunk 5]
```diff
@@ -38,6 +39,16 @@ from the designated branch from earlier.
...
+
+### Running local changes
+TODO
```
Comment (mortenskoett) [inline README.md:50]: I'm currently re-writing ecp_install + adding another script to make this easy. At least for me.

File: scripts/ann-benchmark.sh [hunk 6]
```diff
@@ -9,13 +9,33 @@ fi
 #if no arguments are supplied, display help
 if [[ $# -eq 0 ]]; then
     echo -e "usage: sudo ./run.sh [commit,name,dataset,earlyhalt]"
+    echo -e "flags:"
+    echo -e "-d | --delete | deletes ann-benchmarks folder if exists"
+    echo -e "-l | --local  | runs ann-benchmarks locally"
```
Comment (mortenskoett) [inline scripts/ann-benchmark.sh:14]: The readme says Running local changes id TODO. How does this flag work then?

File: scripts/ann-benchmark.sh [hunk 7]
```diff
@@ -9,13 +9,33 @@ fi
...
+         DELETE_FLAG=true
+	 continue
+    elif [[ $arg == "-l" || $arg == "--local" ]]; then
+	 LOCAL_FLAG=true
+	 continue
+    fi
+done
+
+if $DELETE_FLAG && [ -d ann-benchmarks ]; then
+    rm -rf ann-benchmarks
```
Comment (mortenskoett) [inline scripts/ann-benchmark.sh:36]: We have to agree where to put this dir.

**Review by mortenskoett — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mortenskoett) [reply README.md:31]: Given the inserted strings in both algo and dockerfile.

**Comment by mortenskoett** [comment]
@fremartini using sudo to run the scripts to be able to plot and create website over data is no longer necessary because of latest very nice changes to ann-benchmark repository.

**Comment by mortenskoett** [comment]
@nimertz Can the nice image in the top become part of the documenation as well? I think it will get lost if it is only shown here on the PR.

File: scripts/ann-benchmark.sh (hunk 6 above)
Comment (fremartini) [reply scripts/ann-benchmark.sh:14]: --local runs a commit in ABs local mode. Running local changes means running changes you currently have on your machine

File: scripts/ann-benchmark.sh (hunk 7 above)
Comment (fremartini) [reply scripts/ann-benchmark.sh:36]: is there a problem having it in the root directory? i think it makes sense here as it is located outside our code

File: README.md (hunk 2 above)
Comment (fremartini) [reply README.md:31]: yes

**Review by fremartini — APPROVED** [review]

File: scripts/ann-benchmark.sh (hunk 7 above)
Comment (mortenskoett) [reply scripts/ann-benchmark.sh:36]: I suppose it is a matter of opinion. Given ann-benchmarks is its own repo, I personally prefer to keep it entirely outside the eCP repo.

It's fine.

**Pull request merged**